### PR TITLE
Fix ListPullRequestsWithCommit option type

### PR DESCRIPTION
**Pull request by RickleAndMortimer**
`ListPullRequestsWithCommit` uses `PullRequestListOptions` as one of it's parameters. This is incorrect because in GitHub's documentation, the endpoint for this function, `repos/%v/%v/commits/%v/pulls`, does not share the same parameters as the ones defined in `ListPullRequestsWithCommit`

https://github.com/google/go-github/blob/96726d8192f34c0d46e7b64b3c400962e8e64898/github/pulls.go#L119-L141

image

This PR simply sets the correct options, ListOptions, for the ListPullRequestWithCommit.

see #2815  for more details

### CODE REVIEW

**Comment by gmlewis** [comment]
Thank you, @RickleAndMortimer !
Please run `gofmt` on the files you edited and push (not force push) the changes to this PR.
See CONTRIBUTING.md for more details.

**Comment by gmlewis** [comment]
Fixes: #2815.

**Review by gmlewis — APPROVED** [review]
Thank you, @RickleAndMortimer !
LGTM.

Awaiting second LGTM+Approval from any other contributor to this repo before merging.

**Review by valbeat — APPROVED** [review]
LGTM!

**Pull request merged**
